prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -0,0 +1,328 @@
+        System.out.println( ret );
+        return ret;
+    }
+
+
+    @FunctionName("createEmailSchedule")
+    @StorageAccount("AzureWebJobsStorage")
+    fun createEmailSchedule(
+        @HttpTrigger(
+            name = "createEmailSchedule",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "email-schedule"
+        ) request: HttpRequestMessage<String?>,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        var user:String? = validateUser( request, context );
+        var ret = request.createResponseBuilder(HttpStatus.CREATED);
+
+        if( user !== null ){
+            WorkflowEngine.databaseAccess.insertEmailSchedule(request.body, user );
+        }
+        else{
+            ret.status(HttpStatus.UNAUTHORIZED )
+        }
+        return ret.build();
+    }
+
+
+    fun validateUser( request: HttpRequestMessage<String?>, context: ExecutionContext ): String? {
+        var jwtToken = request.headers["authorization"] ?: ""
+
+        jwtToken = if (jwtToken.length > 7) jwtToken.substring(7) else ""
+
+        if (jwtToken.isNotBlank()) {
+            try {
+                // get the access token verifier
+                val jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
+                    .setIssuer("https://hhs-prime.okta.com/oauth2/default")
+                    .build()
+                // get it to decode the token from the header
+                val jwt = jwtVerifier.decode(jwtToken)
+                    ?: throw Throwable("Error in validation of jwt token")
+                // get the user name and org
+                return jwt.claims["sub"].toString();
+            }
+            catch (ex: Throwable) {
+                context.logger.log(Level.WARNING, "Error in verification of token", ex)
+                return null
+            }
+        }
+        return null;
+    }
+    
+
+    /**
+     * TODO: Fixme!
+     */
+    private fun getSchedules(): List<String>{
+        return WorkflowEngine.databaseAccess.fetchEmailSchedules();
+    }
+
+    /**
+     * Converts an organization name to an OKTA group name
+     */
+    private fun convertOrgToGroup( org: String ): String {
+        return "DH"+org.replace( "-","_" )
+    }
+
+    /**
+     * Retrieve a list of emails within an organization
+     * 
+     * @params org The organization to fetch emails for
+     * @returns List of emails to send to
+     */
+    private fun getEmails( org: String, context: ExecutionContext ): List<String> {
+
+        var ssws: String? = SecretHelper.getSecretService().fetchSecret("SSWS_OKTA")
+
+        if( ssws == null ){
+            context.logger.warning("Can't find SSWS_OKTA secret")
+            return listOf();
+        }
+        
+        var grp = convertOrgToGroup( org );
+
+        // get the OKTA Group Id
+        var response1 = httpGet( url="https://hhs-prime-admin.okta.com/api/v1/groups?q=${grp}", 
```

Comment:
Magic constants; reuse from `OktaCommands.kt` where possible

Reply:
OktaCommands.kt is in cli package - created my own const at head of file